From 36fc8ac7f89be0557912685f3fd4e6e6d1b2afbd Mon Sep 17 00:00:00 2001
From: Tomas Hrnciar <thrnciar@redhat.com>
Date: Mon, 22 Jun 2026 09:51:56 +0200
Subject: [PATCH] Suppress DeprecationWarning for os.forkpty() on Python 3.15+

CPython 3.15 emits a DeprecationWarning when os.forkpty() (or
os.fork()) is called in a multi-threaded process.

See: https://github.com/pexpect/pexpect/issues/827
---
 pexpect/pty_spawn.py | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/pexpect/pty_spawn.py b/pexpect/pty_spawn.py
index 8e28ca7..fafa677 100644
--- a/pexpect/pty_spawn.py
+++ b/pexpect/pty_spawn.py
@@ -5,6 +5,7 @@ import pty
 import tty
 import errno
 import signal
+import warnings
 from contextlib import contextmanager
 
 import ptyprocess
@@ -312,7 +313,12 @@ class spawn(SpawnBase):
 
     def _spawnpty(self, args, **kwargs):
         '''Spawn a pty and return an instance of PtyProcess.'''
-        return ptyprocess.PtyProcess.spawn(args, **kwargs)
+        with warnings.catch_warnings():
+            warnings.filterwarnings(
+                'ignore', category=DeprecationWarning,
+                message=r'.*multi-threaded.*use of fork.*',
+            )
+            return ptyprocess.PtyProcess.spawn(args, **kwargs)
 
     def close(self, force=True):
         '''This closes the connection with the child application. Note that
-- 
2.54.0

